internal/docstore/dynamodocstore/query.go

```diff
@@ -456,33 +456,39 @@ func (qr *queryRunner) queryPlan() string {
 	return "Table"
 }
 
-// TODO(shantuo): use BatchWrite.
 func (c *collection) RunDeleteQuery(ctx context.Context, q *driver.Query) error {
 	q.FieldPaths = [][]string{{c.partitionKey}}
 	if c.sortKey != "" {
 		q.FieldPaths = append(q.FieldPaths, []string{c.sortKey})
 	}
-	iter, err := c.RunGetQuery(ctx, q)
+	qr, err := c.planQuery(q)
 	if err != nil {
 		return err
 	}
-	defer iter.Stop()
+
 	var actions []*driver.Action
+	var startAfter map[string]*dyn.AttributeValue
 	for {
-		doc, err := driver.NewDocument(map[string]interface{}{})
+		items, last, _, err := qr.run(ctx, startAfter)
 		if err != nil {
 			return err
 		}
-		err = iter.Next(ctx, doc)
-		if err == io.EOF {
-			break
+		for _, item := range items {
+			doc, err := driver.NewDocument(map[string]interface{}{})
+			if err != nil {
+				return err
+			}
+			if err := decodeDoc(&dyn.AttributeValue{M: item}, doc); err != nil {
+				return err
+			}
+			actions = append(actions, &driver.Action{Kind: driver.Delete, Doc: doc})
 		}
-		if err != nil {
-			return err
+		if last == nil {
+			break
 		}
-		actions = append(actions, &driver.Action{Kind: driver.Delete, Doc: doc})
+		startAfter = last
 	}
-	alerr := c.RunActions(ctx, actions, &driver.RunActionsOptions{})
+	alerr := c.runActionsUnordered(ctx, actions, &driver.RunActionsOptions{Unordered: true})
```

Comment:
Something seems wrong here. Why do we need to set `Unordered: true` if the function is called `runActionsUnordered`?

Reply:
Yeah this is going to be discarded inside `runActionsUnordered`, I was thinking it might use some other API like batch API's but no. I removed it.